src/core/task/index.ts

```diff
@@ -222,12 +221,23 @@ export class Task {
 			this.startTask(task, images)
 		}
 
+		// Initialize telemetry
+		this.initializeTelemetry(historyItem)
+	}
+
+	// Get the current provider from global state to ensure we're using the latest provider
+	private async getCurrentProviderId(): Promise<string> {
+		return (await getGlobalState(this.getContext(), "apiProvider")) as string
+	}
+
+	private async initializeTelemetry(historyItem?: HistoryItem) {
+		const currentProviderId = await this.getCurrentProviderId()
```

Comment:
since we're only ever calling this once, we should prob leave this in the constructor and use the `apiConfiguration` object being passed in so e.g. `telemetryService.captureTaskRestarted(this.taskId, apiConfiguration.apiProvider)`

Reply:
Fair Point I made those changes